lib/davinci_pas_test_kit/client_suite.rb

```diff
@@ -1,89 +1,103 @@
+        end
       end
-    end
 
-    suite_endpoint :post, TOKEN_PATH, TokenEndpoint
-    suite_endpoint :post, SUBMIT_PATH, ClaimEndpoint
-    suite_endpoint :post, INQUIRE_PATH, ClaimEndpoint
-    suite_endpoint :post, FHIR_SUBSCRIPTION_PATH, SubscriptionCreateEndpoint
-    suite_endpoint :get, FHIR_SUBSCRIPTION_INSTANCE_PATH, SubscriptionsTestKit::SubscriptionReadEndpoint
-    suite_endpoint :post, FHIR_SUBSCRIPTION_INSTANCE_STATUS_PATH, SubscriptionStatusEndpoint
-    suite_endpoint :get, FHIR_SUBSCRIPTION_INSTANCE_STATUS_PATH, SubscriptionStatusEndpoint
-    suite_endpoint :post, FHIR_SUBSCRIPTION_RESOURCE_STATUS_PATH, SubscriptionStatusEndpoint
-
-    resume_test_route :get, RESUME_PASS_PATH do |request|
-      request.query_parameters['token']
-    end
+      route(:get, UDAPSecurityTestKit::UDAP_DISCOVERY_PATH, lambda { |_env|
+        UDAPSecurityTestKit::MockUDAPServer.udap_server_metadata(id)
+      })
+      route(:get, SMARTAppLaunch::SMART_DISCOVERY_PATH, lambda { |_env|
+        SMARTAppLaunch::MockSMARTServer.smart_server_metadata(id)
+      })
 
-    resume_test_route :get, RESUME_FAIL_PATH, result: 'fail' do |request|
-      request.query_parameters['token']
-    end
+      suite_endpoint :post, UDAPSecurityTestKit::REGISTRATION_PATH,
+                     UDAPSecurityTestKit::MockUDAPServer::RegistrationEndpoint
+      suite_endpoint :post, UDAPSecurityTestKit::TOKEN_PATH, MockUdapSmartServer::TokenEndpoint
+
+      suite_endpoint :post, SUBMIT_PATH, ClaimEndpoint
+      suite_endpoint :post, SESSION_SUBMIT_PATH, ClaimEndpoint
+      suite_endpoint :post, INQUIRE_PATH, ClaimEndpoint
+      suite_endpoint :post, SESSION_INQUIRE_PATH, ClaimEndpoint
+      suite_endpoint :post, FHIR_SUBSCRIPTION_PATH, SubscriptionCreateEndpoint
+      suite_endpoint :post, SESSION_FHIR_SUBSCRIPTION_PATH, SubscriptionCreateEndpoint
+      suite_endpoint :get, FHIR_SUBSCRIPTION_INSTANCE_PATH, SubscriptionsTestKit::SubscriptionReadEndpoint
+      suite_endpoint :get, SESSION_FHIR_SUBSCRIPTION_INSTANCE_PATH, SubscriptionsTestKit::SubscriptionReadEndpoint
+      suite_endpoint :post, FHIR_SUBSCRIPTION_INSTANCE_STATUS_PATH, SubscriptionStatusEndpoint
+      suite_endpoint :post, SESSION_FHIR_SUBSCRIPTION_INSTANCE_STATUS_PATH, SubscriptionStatusEndpoint
+      suite_endpoint :get, FHIR_SUBSCRIPTION_INSTANCE_STATUS_PATH, SubscriptionStatusEndpoint
+      suite_endpoint :get, SESSION_FHIR_SUBSCRIPTION_INSTANCE_STATUS_PATH, SubscriptionStatusEndpoint
+      suite_endpoint :post, FHIR_SUBSCRIPTION_RESOURCE_STATUS_PATH, SubscriptionStatusEndpoint
+      suite_endpoint :post, SESSION_FHIR_SUBSCRIPTION_RESOURCE_STATUS_PATH, SubscriptionStatusEndpoint
+
+      resume_test_route :get, RESUME_PASS_PATH do |request|
+        request.query_parameters['token']
+      end
+
+      resume_test_route :get, RESUME_FAIL_PATH, result: 'fail' do |request|
+        request.query_parameters['token']
+      end
```

Comment:
Getting a little nitpicky so feel free to pass on this, but consider renaming these query params to `'session_identifier'` or something now that they're not an access token

Reply:
I get worried about renaming these since having them be consistent across test kits is helpful, else reusing attestation and other wait tests can get difficult. So I'd prefer not to change it, but a larger effort to standardize across all test kits could be worthwhile. Not sure whether that's feasible at this point in the project.

Reply:
fair enough, fine with me